### PR TITLE
sighup-srh test + "_" match for compose

### PR DESCRIPTION
**Pull request by zeeshanlakhani**

### CODE REVIEW

File: examples/srv6-sighup-flow/src/main.rs [hunk 1]
```diff
@@ -115,8 +118,46 @@ where
...
+                                 ProtocolNumbers::Ipv6Route => |group| {
+                                     group.map(|p| {
+                                         let srh = p.parse::<SegmentRouting<Ipv6>>()?;
+                                         warn!("SR {}", srh);
+                                         Ok(srh.deparse())
+                                     })
+                                 }
+                                 // _ => |group| {
+                                 //     group.for_each(|_p| Ok(()))
+                                 // }
```
Comment (drunkirishcoder) [inline examples/srv6-sighup-flow/src/main.rs]: Yeah the difficulty is the look up later. Essentially you want to be able to override the default catch-all behavior of dropping all the unmatched packets with a different catch-all arm. Not that obvious how with the hashmap. To hack around this, we can fall back to using an artificial discriminator like `1, 2, 3...` in the selector to kinda regroup.

Comment (zeeshanlakhani) [reply examples/srv6-sighup-flow/src/main.rs]: @drunkirishcoder did take the `Option` approach, but still hidden from the user. see what you think.

File: framework/src/operators/groupby_batch.rs [hunk 2]
```diff
@@ -92,7 +100,13 @@ where
...
+            $map.insert(Some($key), Box::new(|$arg| Box::new($body)));
+        )*
+    }};
+    ($map:ident, $($key:expr => |$arg:tt| $body:block),*,_ => |$_arg:tt| $_body:block) => {{
+        $(
+            $map.insert(Some($key), Box::new(|$arg| Box::new($body)));
         )*
-    }}
+            $map.insert(None, Box::new(|$_arg| Box::new($_body)));
+    }};
```
Comment (drunkirishcoder) [inline framework/src/operators/groupby_batch.rs]: Can this be cleaned up? looks messy.

File: framework/src/operators/groupby_batch.rs [hunk 3]
```diff
@@ -67,13 +67,21 @@ where
...
+                    match self.groups.get_mut(&Some(key)) {
                         Some(group) => {
                             self.producer.enqueue(packet);
                             group.next().unwrap()
                         }
-                        // can't find the group, drop the packet
-                        None => Err(PacketError::Drop(packet.mbuf())),
+                        None => {
+                            // handle `_` pass-through case
+                            if let Some(group) = self.groups.get_mut(&None) {
```
Comment (drunkirishcoder) [inline framework/src/operators/groupby_batch.rs]: also would be nice to avoid this extra lookup.

Comment (zeeshanlakhani) [reply framework/src/operators/groupby_batch.rs]: @drunkirishcoder separate pattern match, you mean? If so, yes. Still, we’d needs a contains_key check if None is in the map though.

File: framework/src/operators/groupby_batch.rs (hunk 2 above)
Comment (zeeshanlakhani) [reply framework/src/operators/groupby_batch.rs]: ?

Comment (zeeshanlakhani) [reply framework/src/operators/groupby_batch.rs]: Underscores the issue?

File: examples/srv6-sighup-flow/check.sh [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 TEST_NAME=srv6-sighup-flow
-PORT_OPTIONS1="dpdk:eth_pcap0,rx_pcap=data/srv6_tcp.pcap,tx_pcap=/tmp/out.pcap"
 
 C='\033[1;34m'
 NC='\033[0m'
 
 echo -e "${C}RUNNING: $TEST_NAME${NC}"
 
+PORT_OPTIONS1="dpdk:eth_pcap0,rx_pcap=data/srv6_tcp.pcap,tx_pcap=/tmp/out.pcap"
+PORT_OPTIONS2="dpdk:eth_pcap0,rx_pcap=data/icmpv4.pcap,tx_pcap=/tmp/out.pcap"
```
Comment (drunkirishcoder) [inline examples/srv6-sighup-flow/check.sh]: don't need this anymore?

**Pull request merged**